build-tools/packages/build-cli/src/library/packageExports.ts

```diff
@@ -155,7 +155,7 @@ function findTypesPathsMatching<TOutKey>(
  * mapNode10CompatExportPathToData, map of compat file path to Node16 path.
  */
 export function queryTypesResolutionPathsFromPackageExports<TOutKey>(
-	packageJson: PackageJson,
+	packageJson: Pick<PackageJson, "exports">,
```

Comment:
This makes it more clear what data this code works on, and should make it simpler to test.